extension/src/config/init/ensure-resources.js

```diff
@@ -4,16 +4,23 @@ const { ensureApiExtensions } = require('./ensure-api-extensions')
 const ctpClientBuilder = require('../../ctp/ctp-client')
 const config = require('../../config/config')
 
+const ctpClient = ctpClientBuilder.get()
 
-function ensureResources () {
-  const ctpClient = ctpClientBuilder.get()
+function ensureCustomTypes () {
   return Promise.all([
     ensurePaymentCustomType(ctpClient),
-    ensureInterfaceInteractionCustomType(ctpClient),
+    ensureInterfaceInteractionCustomType(ctpClient)
+  ])
+}
+
+function ensureResources () {
+  return Promise.all([
+    ensureCustomTypes(),
     ensureApiExtensions(ctpClient, config.load().apiExtensionBaseUrl)
```

Comment:
Looking at this it seems that there might be a case where integration would run into unhandled TypeError [here](https://github.com/commercetools/commercetools-adyen-integration/blob/master/extension/src/validator/validator-builder.js#L94) if custom was not defined on the payment object. Which might be the case when API extension is created and kicks in before types has been created or when the user of the integration simply forgets to set custom fields on the payment. I guess the issue is not really related to the order of creation here but rather of the linked implementation which misses additional checks?

Reply:
Yes, I agree that the hasMerchantReference check should do additional checks to ensure those objects exist. Would you like me to quickly fix that in a separate PR? I don't think this code specifically is affected by that, rather that it was something missed in the PR of the initial implementation of the merchantReference custom field.

Reply:
Sure, If you have time for that :)

Reply:
Done, PR here #70 :)